### PR TITLE
adds logger and a validate lib

### PR DESCRIPTION
**Pull request by gabeweaver**
# Why
Because it’s useful to have a generic validation lib…and logger is
preferred by some people over redux dev tools.

# What
* Adds logger to redux middleware
* Adds lib directory with some validation functions

### CODE REVIEW

File: universal/containers/AppContainer.js [hunk 1]
```diff
@@ -11,6 +11,7 @@ class AppContainer extends React.Component {
 
   componentDidMount () {
     const { history, store } = this.props
+    console.log(store)
```
Comment (squaresurf) [inline universal/containers/AppContainer.js]: Did you mean to leave this? I can't tell if it's related to the new `redux-logger`.

Comment (jasontwong) [inline universal/containers/AppContainer.js]: This should be removed.

File: universal/store/createStore.js [hunk 2]
```diff
@@ -1,6 +1,9 @@
 import { applyMiddleware, compose, createStore } from 'redux'
 import thunk from 'redux-thunk'
 import makeRootReducer from './reducers'
+import createLogger from 'redux-logger'
+
+const logger = createLogger()
```
Comment (bbrock25) [inline universal/store/createStore.js]: we should probably create the logger within that if statement so that this isn't executed when its not needed.

Comment (gabeweaver) [inline universal/store/createStore.js]: @bbrock25 i went ahead and did that but it doesnt matter so long as the const is defined above where i call it in the scope....it doesn't get executed until it's used though, in this case in the push()

Am i missing something? Either way...it's in there now and changed to be defined within the __DEBUG__ block

**Pull request merged**